test/framework/src/main/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -272,6 +281,11 @@ private int hashCode(String path) {
             }
 
             private void maybeIOExceptionOrBlock(String blobName) throws IOException {
+                if (INDEX_LATEST_BLOB.equals(blobName)) {
```

Comment:
We have some annoying tests where random failures in `index.latest` would get counted via `incrementAndGetFailureCount` but because they now do not break the snapshot, the test would fail because random IOException must always mean failed snapshot in those tests => this seemed like the simplest fix to that.